src/saveload/marker.rs

```diff
@@ -362,11 +365,38 @@ pub trait MarkerAllocator<M: Marker>: Resource {
 
 /// Basic marker implementation usable for saving and loading, uses `u64` as
 /// identifier
-#[derive(Derivative, Serialize, Deserialize)]
-#[derivative(Clone, Copy, Debug, Hash, PartialEq, Eq)]
+#[derive(Serialize, Deserialize)]
 #[repr(transparent)]
 pub struct SimpleMarker<T: ?Sized>(u64, #[serde(skip)] PhantomData<T>);
 
+impl<T: ?Sized> Clone for SimpleMarker<T> {
+    fn clone(&self) -> Self {
+        *self
+    }
+}
+
+impl<T: ?Sized> Copy for SimpleMarker<T> {}
+
+impl<T: ?Sized> PartialEq for SimpleMarker<T> {
+    fn eq(&self, other: &Self) -> bool {
+        self.0 == other.0
+    }
+}
+
+impl<T: ?Sized> Eq for SimpleMarker<T> {}
+
+impl<T: ?Sized> Hash for SimpleMarker<T> {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.0.hash(state);
+    }
+}
+
+impl<T: ?Sized> Debug for SimpleMarker<T> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "SimpleMarker({:?}, PhantomData)", self.0)
```

Comment:
Think it would be better to use [`Formatter::debug_tuple`](https://doc.rust-lang.org/std/fmt/struct.Formatter.html#method.debug_tuple) -- I think it handles writing for both normal debug and pretty-print debug.
This *might* work:

```rust
fmt.debug_tuple("SimpleMarker")
    .field(&self.0)
    .field(&format_args!("_"))
    .finish()
```